src/compiler/scala/tools/nsc/reporters/ConsoleReporter.scala

```diff
@@ -25,9 +25,11 @@ class ConsoleReporter(val settings: Settings, val reader: BufferedReader, val wr
   def doReport(pos: Position, msg: String, severity: Severity): Unit = display(pos, msg, severity)
 
   override def finish(): Unit = {
-    import reflect.internal.util.StringOps.countElementsAsString
-    if (hasWarnings) echo(countElementsAsString(warningCount, WARNING.toString.toLowerCase))
-    if (hasErrors)   echo(countElementsAsString(errorCount, ERROR.toString.toLowerCase))
+    if (!settings.nowarn) {
+      import reflect.internal.util.StringOps.countElementsAsString
+      if (hasWarnings) echo(countElementsAsString(warningCount, WARNING.toString.toLowerCase))
+      if (hasErrors)   echo(countElementsAsString(errorCount, ERROR.toString.toLowerCase))
```

Comment:
the error summary should probably stay even with `-nowarn`?

Reply:
You mean error reporting? Yes, I had a second thought about that. I'll update to retain it. That was an example of hitting my time budget and patience budget.